classes/CartRule.php

```diff
@@ -1982,7 +1982,7 @@ public static function autoAddToCart(?Context $context = null, bool $useOrderPri
 		LEFT JOIN ' . _DB_PREFIX_ . 'cart_rule_country crco ON cr.id_cart_rule = crco.id_cart_rule
 		WHERE cr.active = 1
 		AND cr.code = ""
-		AND cr.quantity > 0
+		AND (cr.quantity > 0 OR cr.quantity is null)
```

Comment:
Did you search in other files (especially legacy ones) if these two fields are used? especially in conditions?